### DISCUSSION TITLE
Unable to get AWS AppConfig configuration file with boto3_client passed to AppConfigStore

### DISCUSSION THREAD

**Discussion by Rogalek** [aws-powertools/powertools-lambda-python#5511] [Q&A] [answered]
Hello

When I am trying to run `feature_flags.get_enabled_features()` I am getting this error:
```
[DEBUG]	2024-11-04T14:53:42.210Z	cab62e2e-1609-4121-a058-6fd3eec9c7e2	Failed to fetch feature flags from store, returning empty list, reason=Unable to get AWS AppConfig configuration file
````

I am trying to run this code:
```
    app_config = AppConfigStore(
        environment=ENV,
        application=APPCONFIG_APPLICATION_NAME,
        name=configuration_profile,
        sdk_config=Config(region_name=APPCONFIG_REGION),
        boto3_client=restricted_client(service_name="appconfig", region_name=APPCONFIG_REGION),
    )
    feature_flags = FeatureFlags(store=app_config, logger=logger)
    features = feature_flags.get_enabled_features(context=ctx)
 ```
 
 When I use this restricted_client (its a my own class which inherits from boto3) for example to:
 `restricted_client.list_configuration_profiles(ApplicationId=APPCONFIG_APPLICATION_ID)`  - It is working perfectly but I cannot evaluate flags.
 
 Can someone help me? What I do wrong?

**Comment by leandrodamascena** [comment]
Hi @Rogalek sorry for the super late reply. Let me try to run this code and I'll get back to you soon.

**Reply by Rogalek** [reply]
Thank you - waiting for your reply 🙏

**Reply by leandrodamascena** [reply, marked as answer]
Hi @Rogalek! I finally found some time to inspect it. The problem is that you are using an instance of `appconfig` boto3.client, when in reality you need to use `appconfigdata` because this is the client that can communicate with the AppConfig Data Plane and query its values ​​stored in your application and environment. We [mention](https://docs.powertools.aws.dev/lambda/python/latest/utilities/feature_flags/#appconfig) that in our documentation.

`restricted_client(service_name="appconfig", region_name=APPCONFIG_REGION)`

should be

`restricted_client(service_name="appconfigdata", region_name=APPCONFIG_REGION)`

Please let me know if this works.

**Reply by Rogalek** [reply]
thank you - this is it - I missed that.

**Reply by leandrodamascena** [reply]
Nice! Please ping if you need any additional help.
